scripts/main/view.js

```diff
@@ -440,6 +450,26 @@ view.album = {
 				);
 		},
 
+		/**
+		 * Attaches a datasource to the photo view.
+		 *
+		 * This frees us from re-determining the current data source every
+		 * time the layout needs to be updated.
+		 * The data source is an object with an attribute `json` which in
+		 * turn may hold an array `photos` of `Photo`.
+		 * Currently, this can either be {@link album} or {@link search}.
+		 * Note, we deliberately do not take a direct reference to the JSON
+		 * object ot the array of photos such that we do not accidentally
+		 * keep a pointer on an outdated array of photos when {@link album} or
+		 * {@link search} gets cleared.
+		 *
+		 * @param {?{json: ?{photos: Photo[]}}} photoDataSource
+		 */
+		setPhotoDataSource: function (photoDataSource) {
+			view.album.content.photoDataSource = photoDataSource;
+			setTimeout(() => view.album.content.justify(), 0);
+		},
+
```

Comment:
Do we need this method at all? Search now sets `album` to its result as well...

Reply:
Sorry, here's what I meant:

Some time back, you refactored the search code so that search results were stored in a new location: `search.json`. I was worried that too many places may depend on the currently displayed photos being stored in`album.json`, so I requested that you store the search results there as well, effectively creating a virtual "search smart album", so to speak. You implemented that.

Therefore, I don't think there's any reason anymore to decide whether to use `album` or `search` as the source of photos: in any situation where `search` would be appropriate, the same result is also stored in `album`. Therefore, always loading from `album` should be fine. We shouldn't need a method that allows to specify a different source.

Does that make sense?

Reply:
Yes that makes total sense. I cannot even remember that I implemented that change you are speaking about. So, what you are saying is that the previous lines of code like
```js
if (visible.album()) view.album.content.justify(album.json ? album.json.photos : []);
if (visible.search()) view.album.content.justify(search.json.photos);
```
have already been unnecessary and `view.album.content.justify()` would have sufficed.

That's would be great. I am going to check that and probably change it. (How embarrassing for me to not remembering that.)

Reply:
@kamil4 You were absolutely right.

Although I coded this

https://github.com/LycheeOrg/Lychee-front/blame/a89efe2b59da434dcf3a933d5a3318a4c1edae5f/scripts/main/search.js#L58-L69

myself, I could not remember.

Fixed it in https://github.com/LycheeOrg/Lychee-front/pull/334/commits/fcb3cfb3775e31c01b09e426296f49c6ac853ea0.

Reply:
No worries; I remembered it for you :smiley:.